client/state/happychat/middleware.js

```diff
@@ -129,10 +132,19 @@ export const connectChat = ( connection, { getState, dispatch } ) => {
 
 			return sign( { user, session_id } );
 		} )
-		.then( ( { jwt } ) => connection.open( user.ID, jwt, locale ) )
+		.then( ( { jwt } ) => connection.open( user.ID, jwt, locale, groups ) )
 		.catch( e => debug( 'failed to start happychat session', e, e.stack ) );
 };
 
+
+export const updateChatPreferences = ( connection, { getState }, siteId ) => {
+	const state = getState();
+	const locale = getCurrentUserLocale( state );
+	const groups = getGroups( state, siteId );
+
+	connection.preferences( locale, groups );
```

Comment:
This should check first if there's an open connection established. When there isn't, this error can occur:

<img width="532" alt="screen shot 2017-06-05 at 6 57 06 pm" src="https://cloud.githubusercontent.com/assets/518059/26808220/f4da67d4-4a20-11e7-9d78-060c83889cdb.png">

This is because the `connection` class doesn't set the `openSocket` Promise until `open()` is called — so if any of its member functions are called before `open()` this error happens.

We should figure out a long-term solution for all functions, but it's especially problematic in this PR because the `HELP_SELECTED_SITE` action can happen before Happychat forms a connection.